x-pack/plugins/uptime/public/components/overview/alerts/monitor_expressions/filters_expression_select.tsx

```diff
@@ -123,43 +122,60 @@ export const FiltersExpressionsSelect: React.FC<Props> = ({
     (curr) => curr.selectedItems.length > 0 || newFilters?.includes(curr.fieldName)
   );
 
+  const indexPattern = useIndexPattern();
+
   return (
     <>
-      {filtersToDisplay.map(({ description, value, ...item }) => (
-        <EuiFlexGroup key={item.id}>
-          <EuiFlexItem>
-            <FilterPopover
-              {...item}
-              btnContent={
-                <EuiExpression
-                  aria-label={item['aria-label']}
-                  color={'secondary'}
-                  data-test-subj={'uptimeCreateStatusAlert.' + item.id}
-                  description={description}
-                  value={value}
-                  onClick={() => setIsOpen({ ...isOpen, [item.id]: !isOpen[item.id] })}
+      {filtersToDisplay.map(
+        ({ description, id, title, value, fieldName, ariaLabel, selectedItems }) => (
+          <EuiFlexGroup key={id}>
+            <EuiFlexItem>
+              {indexPattern && (
+                <FieldValueSuggestions
+                  filters={[]}
+                  key={fieldName}
+                  indexPatternTitle={indexPattern.title}
+                  sourceField={fieldName}
+                  label={title}
+                  onChange={(vals) => {
+                    onFilterFieldChange(fieldName, vals);
+                  }}
+                  selectedValue={selectedItems}
+                  button={
+                    <EuiExpression
+                      aria-label={ariaLabel}
+                      color={'secondary'}
+                      data-test-subj={'uptimeCreateStatusAlert.' + id}
+                      description={description}
+                      value={value}
+                      onClick={() => setIsOpen({ ...isOpen, [id]: !isOpen[id] })}
```

Comment:
I'm not sure what the weight is of `EuiExpression`, but we could perhaps benefit from memoization here as well, especially if updates to this component then also trigger an update to `FieldValueSuggestions`.

Reply:
```suggestion
                      onClick={() => setIsOpen((prevState) => ({ ...prevState, [id]: !prevState[id] })}
```
Can also use functional setState here, which, if you do decide to memoize, would cut down on hook dependencies.